File: rie/python/2.py
#coding=utf-8
num=input('请输入一个不多于五位的正整数:')
if num<=9:
	print'它是一位数'
elif num<=99:
	print'它是两位数'
elif num<=999:
	print '它是三位数'
elif num<=9999:
	print'它是四位数'
else:
	print'它是五位数'
a=num//10000
b=(num-a*10000)//1000
c=(num-a*10000-b*1000)//100
d=(num-a*10000-b*1000-c*100)//10
e=num-a*10000-b*1000-c*100-d*10
print '万位:',a
print'千位:',b
print'百位:',c
print'十位:',d
print'个位:',e
print'它的逆序为：',e,d,c,b,a

File: rie/yucl/2.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#define NUM 3

typedef struct names 
{
	char first[20];
	char last[20];
} NA, *PNA;

void input (NA [], int );
void output (NA [], int);
int mycomp (const void *, const void *);

int main (void)
{
	NA arr[NUM];
	input (arr, NUM);
	puts ("Before Sorted: ");
	output (arr, NUM);
	qsort (arr, NUM, sizeof (NA), mycomp);
	puts ("After Sorted: ");
	output (arr, NUM);
	return 0;
}

void input (NA arr[], int n)
{
	register int index;
	printf ("Enter %d names:\n", n);
	
	for (index = 0; index < n; index++)
	{
		printf ("NO.%d's name: ", index+1);
		scanf ("%s %s", arr[index].first, arr[index].last);
		putchar ('\n');
	}
}

void output (NA arr[], int n)
{
	register int index;
	for (index = 0; index < n; index++)
		printf ("NO.%d: %s %s\n", index+1, arr[index].first, arr[index].last);
}

int mycomp (const void * p1, const void * p2)
{
	int temp;
	const NA * a1 = (const void *)p1;
	const NA * a2 = (const void *)p2;
	temp = strcmp (a1->last, a2->last);
	
	if (temp != 0)
		return temp;
	else 
		return strcmp (a1->first, a2->first);
}
